Packs/Core/Integrations/CortexCoreIR/CortexCoreIR.py

```diff
@@ -183,6 +189,103 @@ def get_asset_details_command(client: Client, args: dict) -> CommandResults:
     )
 
 
+def parse_expiration_date(expiration: Optional[str]) -> Optional[Union[int, str]]:
+    """
+     Converts relative expiration strings / numbers to epoch milliseconds or returns 'Never'.
+
+    Args:
+        expiration Optional[str]: The input from the command argument
+
+    Returns:
+        Optional[int, str]: The valiue that represent the expiration date of the IOC rule:
+            None: the rule get a default value.
+            str: "Never" - The rule has no expiration date.
+            int: epoch milliseconds of the expiration date.
+    """
+
+    if not expiration:
+        return None
+    if expiration == "Never":
+        return "Never"
+
+    def convert_datetime_to_epoch_milli(dt: datetime) -> int:
+        return int(dt.timestamp() * 1000)
+
+    def is_relative_time_format(s: str) -> bool:
+        """
+        Returns True if the input matches a relative time format:
+        'N minutes', 'N hours', 'N days', 'N weeks', 'N months', 'N years'
+        """
+        if not isinstance(s, str):
+            return False
+
+        pattern = r"^\s*\d+\s+(minutes|hours|days|weeks|months|years)\s*$"
+        return bool(re.match(pattern, s, flags=re.IGNORECASE))
+
+    now_epoch_milli = convert_datetime_to_epoch_milli(get_current_time())
+    try:
```

Comment:
Also you might consider doing something like this:
```

    try:
        dt = arg_to_datetime(expiration)
    except ValueError:
        return expiration  # Invalid input, pass through

    if dt:
        if re.fullmatch(r"\s*\d+\s+(minutes|hours|days|weeks|months|years)\s*", expiration, re.IGNORECASE):
            now = date_to_timestamp(get_current_time())
            delta = now - date_to_timestamp(dt)
            return now + delta
        else:
            return date_to_timestamp(dt)
```
Clean the code a bit - very hard to follow